plugins/inputs/smart/smart.go

```diff
@@ -373,6 +391,37 @@ func gatherDisk(acc telegraf.Accumulator, timeout internal.Duration, usesudo, co
 				}
 			}
 		}
+
+		if m.ErrorCounterLog {
+			ecl := errorCounterLog.FindStringSubmatch(line)
+			if len(ecl) > 1 {
```

Comment:
I am not sure if I follow, do you mean to check for number of match groups like code below ?

```
if len(ecl) == 8 {
```

Reply:
Correct

Reply:
I agree that it's more explicit and readable, changed. However it's just one simple regex and it would match only as a whole with 8 match groups or not match at all.